### PR TITLE
add categorize toggle to inventory

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
```SUMMARY: Features "Adds toggle key in inventory to show categorized items"```

#### Purpose of change
i've been hearing a lot from people about how they want the categorized view instead of everything being shown under what item it's contained in, so i added a toggle button in the main inventory screen.

#### Describe the solution

Adjusts the inventory columns when hitting the toggle so the own_inv-column is used, and adds it via add_entry so categories are applied. pressing the toggle again will put it back to how it was.

#### Testing

Have a decently sized inventory and press the toggle button.

#### Additional context

screenshots to follow when this comes out of draft.

### CODE REVIEW

**Comment by KorGgenT** [comment]
inv

File: msvc-full-features/enc_temp_folder/41d0c12c7c36ae135e7872bfb7afcaa8/inventory_ui.cpp [hunk 1]
```diff
@@ -0,0 +1,2375 @@
+#include "inventory_ui.h"
```
Comment (anothersimulacrum) [inline msvc-full-features/enc_temp_folder/41d0c12c7c36ae135e7872bfb7afcaa8/inventory_ui.cpp:1]: This doesn't seem like it belongs.

Comment (KorGgenT) [reply msvc-full-features/enc_temp_folder/41d0c12c7c36ae135e7872bfb7afcaa8/inventory_ui.cpp:1]: woops!

**Comment by CodeBandit** [comment]
I hope this gets merged asap, UI clarity is a big deal

**Comment by kevingranade** [comment]
One nit, the "old" way of displaying this should be the default, the new way of drawing the inventory is useful for shuffling items around, but the *goal* is for that to be rarely necessary.

**Comment by KorGgenT** [comment]
ah. i'll have to do some shuffling of the initialization of the menu then.

**Comment by KorGgenT** [comment]
image
just did a rebase, works great with the stack thing.

**Pull request merged**